crates/edr_napi/index.d.ts

```diff
@@ -461,6 +461,15 @@ export interface TestContract {
    */
   libraries: Array<string>
 }
+/**
+ * Executes Solidity tests.
+ *
+ * The function will return as soon as test execution is started.
+ * The progress callback will be called with the results of each test suite.
+ * It is up to the caller to track how many times the callback is called to
+ * know when all tests are done.
+ */
+export function runSolidityTests(test_suites: Array<TestSuite>, gas_report: boolean, progress_callback: (result: SuiteResult) => void): void
```

Comment:
Consideration: As this is spawning a background process, it might be good to indicate this in the function name. The only thing I can come up with is `spawnSolidityTestRunner`.

Up to you whether you think this makes sense. The docs also outline this, in case the name is not self-explanatory.

Reply:
I'm ok with having this low-level interface being callback based, as we can then wrap it in something more idiomatic for js. But it should have a simple callback indicating the completion of the process. 

I know we discussed leaving that responsibility to the user, but I think that would be less flexible/evolvable. E.g. if we implement a feature in edr that skips parts of the tests, every consumer would have to adapt the "counting the results until finish" code.

Note that the completion callback doesn't need to accumulate all the results. It could probably be `() => void`.

Reply:
> But it should have a simple callback indicating the completion of the process.

I like the idea and I can make that change, but tbh I'm a bit wary of doing it for the following reasons:

1. For now, the only users of this interface are in the EDR repo and there would be no usage of the completion callback.
3. We ran into the problem fixed by this PR, because we tried to anticipate usage patterns and provide a more ergonomic interface. This tells me that it's too early to polish the interface.

So I'd just add the completion callback to the interface design doc if that's ok?